rcl_lifecycle/include/rcl_lifecycle/rcl_lifecycle.h

```diff
@@ -0,0 +1,110 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL_LIFECYCLE__RCL_LIFECYCLE_H_
+#define RCL_LIFECYCLE__RCL_LIFECYCLE_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#ifndef __cplusplus
+  #ifndef bool
+    #define bool int
+    #define true 1
+    #define false 0
+  #endif
+#endif
+
+#include <rcl_lifecycle/visibility_control.h>
+#include <rcl_lifecycle/data_types.h>
+
+RCL_LIFECYCLE_PUBLIC
+rcl_lifecycle_state_machine_t
+rcl_lifecycle_get_zero_initialized_state_machine();
+
+RCL_LIFECYCLE_PUBLIC
+rcl_ret_t
+rcl_lifecycle_state_machine_init(rcl_lifecycle_state_machine_t * state_machine,
```

Comment:
nitpick: please wrap all parameters or none of them, but please do not wrap all but the first (this occurs throughout this file)